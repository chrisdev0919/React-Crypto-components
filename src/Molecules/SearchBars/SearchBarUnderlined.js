import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Input } from '../../Atoms/Inputs'
import { Icon } from '../../Atoms/Icons'

const BaseSearchBarUnderlined = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid ${props => props.theme['gray-2']};
  background-color: transparent;
`
const IconContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 60px;
  height: 40px;
  background-color: inherit;
`

const SearchBarUnderlined = ({ children, value, handleChange, handleClick, ...rest }) => (
  <BaseSearchBarUnderlined>
    <Input backgroundColor='transparent' borderColor='transparent' value={value} onChange={handleChange} {...rest} />
    <IconContainer>
      <Icon name='search-filled' size='20px' selectable onClick={handleClick} />
    </IconContainer>
  </BaseSearchBarUnderlined>
)

SearchBarUnderlined.propTypes = {
  value: PropTypes.any,
  handleChange: PropTypes.func,
  handleClick: PropTypes.func
}

export default SearchBarUnderlined
