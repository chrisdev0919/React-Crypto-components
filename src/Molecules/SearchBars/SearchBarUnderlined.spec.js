import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SearchBarUnderlined from './SearchBarUnderlined'

describe('SearchBarUnderlined component', () => {
  it('renders correctly', () => {
    const component = shallow(<SearchBarUnderlined />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with value', () => {
    const component = shallow(<SearchBarUnderlined value='My search' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with handleChange', () => {
    const component = shallow(<SearchBarUnderlined handleChange={() => console.log('change')} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with handleClick', () => {
    const component = shallow(<SearchBarUnderlined handleClick={() => console.log('click')} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
