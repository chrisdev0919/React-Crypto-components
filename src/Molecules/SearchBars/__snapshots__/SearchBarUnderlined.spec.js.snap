// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarUnderlined component renders correctly 1`] = `
<styled.div>
  <Input
    backgroundColor="white"
    borderColor="transparent"
    color="gray-5"
    disabled={false}
    type="text"
    width="100%"
  />
  <styled.div>
    <Icon
      color="gray-5"
      name="search-filled"
      selectable={true}
      size="20px"
      weight={400}
    />
  </styled.div>
</styled.div>
`;

exports[`SearchBarUnderlined component renders correctly with handleChange 1`] = `
<styled.div>
  <Input
    backgroundColor="white"
    borderColor="transparent"
    color="gray-5"
    disabled={false}
    onChange={[Function]}
    type="text"
    width="100%"
  />
  <styled.div>
    <Icon
      color="gray-5"
      name="search-filled"
      selectable={true}
      size="20px"
      weight={400}
    />
  </styled.div>
</styled.div>
`;

exports[`SearchBarUnderlined component renders correctly with handleClick 1`] = `
<styled.div>
  <Input
    backgroundColor="white"
    borderColor="transparent"
    color="gray-5"
    disabled={false}
    type="text"
    width="100%"
  />
  <styled.div>
    <Icon
      color="gray-5"
      name="search-filled"
      onClick={[Function]}
      selectable={true}
      size="20px"
      weight={400}
    />
  </styled.div>
</styled.div>
`;

exports[`SearchBarUnderlined component renders correctly with value 1`] = `
<styled.div>
  <Input
    backgroundColor="white"
    borderColor="transparent"
    color="gray-5"
    disabled={false}
    type="text"
    value="My search"
    width="100%"
  />
  <styled.div>
    <Icon
      color="gray-5"
      name="search-filled"
      selectable={true}
      size="20px"
      weight={400}
    />
  </styled.div>
</styled.div>
`;
